### PR TITLE
Issue- 178: E2E tests fixed and Travis support added

### LINKED ISSUE #149: Fix broken e2e tests
After moving to the latest Operator-SDK End to End test are broken.

### PR DESCRIPTION
**Pull request by Prabhaker24**
Signed-off-by: prabhaker24 <prabhaker.saxena@dell.com>

### Change log description
Because of the latest operator-SDK changes End to end tests were broken which got fixed in this pr and support for running e2e on Travis over minikube is added.

### Purpose of the change
fixes #149  #178 #181 

### What the code does
It fixes the e2e tests which got broken because of later operator SDK changes and adds support to run those tests over Travis over minikube, which will help in testing the code once a pr is raised. As we are making changes in the makefile and adding the username and password in the makefile so we don't need to add to the Travis setting anymore and we are able to run e2e tests on travis for fork branches, thus this pr solves issue 181 also.

### How to verify it
It can be tested by raising a pr which would, in turn, trigger a Travis build which runs e2e tests.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,13 +13,15 @@ EXPORTER_NAME=zookeeper-exporter
...
 REPO=pravega/$(PROJECT_NAME)
-TEST_REPO=$(DOCKER_USER)/$(PROJECT_NAME)
+TEST_REPO=testzkop/$(PROJECT_NAME)
 APP_REPO=pravega/$(APP_NAME)
 ALTREPO=emccorp/$(PROJECT_NAME)
 APP_ALTREPO=emccorp/$(APP_NAME)
 VERSION=$(shell git describe --always --tags --dirty | sed "s/\(.*\)-g`git rev-parse --short HEAD`/\1/")
 GIT_SHA=$(shell git rev-parse --short HEAD)
 TEST_IMAGE=$(TEST_REPO)-testimages:$(VERSION)
+DOCKER_TEST_PASS=testzkop@123
```
Comment (pbelgundi) [inline Makefile:23]: I believe with this change we'll be able to run these tests on fork branches as well?

File: test/e2e/multiple_zk_test.go [hunk 2]
```diff
@@ -57,8 +58,9 @@ func testMultiZKCluster(t *testing.T) {
 	defaultCluster.Status.Init()
 	defaultCluster.Spec.Persistence.VolumeReclaimPolicy = "Delete"
 	defaultCluster.ObjectMeta.Name = "zk2"
-	initialVersion := "0.2.5"
-	upgradeVersion := "0.2.7"
+	initialVersion := "0.2.7"
+	upgradeVersion := "0.2.8-rc0"
```
Comment (pbelgundi) [inline test/e2e/multiple_zk_test.go:62]: Could you look at making the `initial` and `upgrade` Versions configurable so we don;t need to change this file everytime we do a new ZK release?

File: test/e2e/multiple_zk_test.go [hunk 3]
```diff
@@ -80,7 +82,7 @@ func testMultiZKCluster(t *testing.T) {
 	defaultCluster.Status.Init()
 	defaultCluster.ObjectMeta.Name = "zk3"
 	defaultCluster.Spec.Persistence.VolumeReclaimPolicy = "Delete"
-
+	defaultCluster.Spec.Replicas = 1
```
Comment (pbelgundi) [inline test/e2e/multiple_zk_test.go:85]: Why reduce replica count from 3 to 1 ?

File: test/e2e/zookeepercluster_test.go [hunk 4]
```diff
@@ -57,9 +57,9 @@ func testZookeeperCluster(t *testing.T) {
 	}
 
 	testFuncs := map[string]func(t *testing.T){
+		"testUpgradeCluster":        testUpgradeCluster,
```
Comment (pbelgundi) [inline test/e2e/zookeepercluster_test.go:60]: Is there a reason for moving this on top?

**Comment by anishakj** [comment]
Currently end to end tests are running from the 2 of the jobs. Either we can limit to one job or at least make end to end tests to run from only one job .

File: .travis.yml [hunk 5]
```diff
@@ -22,14 +39,3 @@ deploy:
     on:
       branch: master
       tags: true
-  - provider: releases
```
Comment (anishakj) [inline .travis.yml:25]: why are we removing these?

File: Makefile (hunk 1 above)
Comment (Prabhaker24) [reply Makefile:23]: Yes, these changes will run e2e test on fork branches as well.

File: test/e2e/multiple_zk_test.go (hunk 3 above)
Comment (Prabhaker24) [reply test/e2e/multiple_zk_test.go]: I had reduced it as the test was taking more than 10 minutes to run and it was getting timed out I found out a way to solve it and have changed it back to 3 again.

File: test/e2e/zookeepercluster_test.go (hunk 4 above)
Comment (Prabhaker24) [reply test/e2e/zookeepercluster_test.go]: There's no specific reason, I have changed it back.

File: .travis.yml (hunk 5 above)
Comment (Prabhaker24) [reply .travis.yml]: I have added it back.

File: Makefile (hunk 1 above)
Comment (pbelgundi) [reply Makefile:23]: In that case, could you please add issue #181 in the "Fixes" section of this PR?

File: test/e2e/zookeepercluster_test.go (hunk 4 above)
Comment (pbelgundi) [reply test/e2e/zookeepercluster_test.go]: cool

File: Makefile (hunk 1 above)
Comment (Prabhaker24) [reply Makefile:23]: sure, I have added it as well.

**Pull request merged**